pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -261,8 +261,11 @@ public void invalidate(String path) {
 
     @Override
     public void refresh(String path) {
-        objCache.synchronous().invalidate(path);
-        objCache.synchronous().refresh(path);
+        // Refresh object of path if only it is cached before.
+        if (objCache.getIfPresent(path) != null) {
```

Comment:
I'm just wondering if there's a possibility for races.

Reply:
> we should move this block into a dedicated method of objCache
> 
> `objCache.synchronous().invalidateAndRefreshIfPresent(path)`

It's a bit of trouble to do this, because objCache is an instance of class `com.github.benmanes.caffeine.cache.AsyncCacheLoader` , which is in a dependent lib.